tests/puppeteer/pages/BO/payment/preferences/index.js

```diff
@@ -27,13 +26,13 @@ module.exports = class Preferences extends BOBasePage {
    */
   async setCurrencyRestriction(paymentModule, valueWanted) {
     await this.waitForVisibleSelector(
-      `${this.euroCurrencyRestrictionsCheckbox.replace('%PAYMENTMODULE', paymentModule)} + i`,
+      `${this.euroCurrencyRestrictionsCheckbox(paymentModule)} + i`,
     );
     const isCheckboxSelected = await this.isCheckboxSelected(
-      this.euroCurrencyRestrictionsCheckbox.replace('%PAYMENTMODULE', paymentModule),
+      this.euroCurrencyRestrictionsCheckbox(paymentModule),
     );
     if (valueWanted !== isCheckboxSelected) {
-      await this.page.click(`${this.euroCurrencyRestrictionsCheckbox.replace('%PAYMENTMODULE', paymentModule)} + i`);
+      await this.page.click(`${this.euroCurrencyRestrictionsCheckbox(paymentModule)} + i`);
```

Comment:
This is the only thing that is bugging me, calling the method inside a `` and ${}
Not sure it's good practice, maybe it would be better to have another method:
```js
this.euroCurrencyRestrictionsCheckboxIndexed = (paymentModule) => `input#form_payment_module_preferences_currency_restrictions_${paymentModule}_0 + i`;
```

@NeOMakinG @PierreRambaud what do you think?

Reply:
That would be way better but well, it's not a big deal at all for me